server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java

```diff
@@ -178,8 +181,15 @@ public void save() {
                     writer.write(recordMap);
                 }
             }
-        } catch (IOException e) {
-            throw new SwProcessException(ErrorType.DATASTORE, "failed to save " + this.tableName, e);
+            builder.success();
+        } catch (Throwable e) {
+            builder.error();
```

Comment:
Mark it with an error when init, and no need to change the try segment?

Reply:
done